bears/general/InvalidLinkBear.py

```diff
@@ -68,13 +68,53 @@ def find_links_in_file(file, timeout, link_ignore_regex):
                                         # set of parenthesis.
                                         # An example can be:
                                         # http://wik.org/Hello_(Adele_song)/200
-            )
-            *)
-                                        # Thus, the whole part above
+            )                           # Thus, the whole part above
                                         # prevents matching of
                                         # Unbalanced parenthesis
+            (
+                (%?[0-9]+)              # A ``%`` sign (0 or 1 times)
+                                        # followed by any number
+                                        # Thus ``%25`` is allowed
+                                        # and ``%s`` isn't allowed
+                                        # This also puts a restriction
+                                        # that the URLs have to end with
+                                        # a number (HTTP Status Code)
+                [^\s()%\'"`<>|\\]+      # Same Regex for Path Name
+            |                           # OR
+                \([^\s()%\'"`<>|\\]*\)  # Same Regex for Path name
+                                        # contained within ()
+                ([^\s()%\'"`<>|\\]*)    # Same Regex for Path Name
+                                        # for including HTTP Status Codes
+            )
             (?<!\.)(?<!,)               # Exclude trailing `.` or `,` from URL
-            """, re.VERBOSE)
+            )                           # This is the first part of the regex
+                                        # that covers URLs with or without
+                                        # placeholders but with necessary
+                                        # HTTP Status Codes at the end
+            |                           # OR (Second part of regex begins here)
+                                        # This covers URLs without any
+                                        # numbers at the end
+                                        # This is essential for URLs
+                                        # with placeholders but
+                                        # without HTTP Status Codes
```

Comment:
But I should let the meaning be intact right ? I will try to re-frame these phrases :+1: